CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
Well, I would say both explicit/implicit approach has advantage/disadvantage over each other on handling multi-select value(s) but using explicit would need to mention special character in value while using api.Get. But on a brighter side it fixes that any/all issue, good job :) - reason why I merged it. However there is a scope of optimization in the current patch for which I will send a PR later.